admin/class-smaily-for-wp-admin.php

```diff
@@ -324,21 +391,17 @@ public function smaily_admin_save() {
 	 * @return array $autoresponder_list List of autoresponders in format [id => title].
 	 */
 	public function get_autoresponders() {
-		global $wpdb;
-
 		// Load configuration data.
-		$table_name = esc_sql( $wpdb->prefix . 'smaily_config' );
-		$config     = (array) $wpdb->get_row( "SELECT * FROM `$table_name` LIMIT 1" );
+		$api_credentials = $this->option_handler->get_api_credentials();
 
-		if ( empty( $config['api_credentials'] ) ) {
+		$credentials_not_valid = empty( $api_credentials['subdomain'] ) || empty( $api_credentials['username'] ) || empty( $api_credentials['password'] );
```

Comment:
Why are you not using `has_credentials` helper here?

Reply:
Since credentials are needed later when making a request. If I would replace it with the helper, API credentials would be loaded twice.

I modified the has_credentials helper to support passing in api_credentials via function parameter and implemented it in 2424fcb

Reply:
Since credentials are needed later when making a request. If I would replace it with the helper, API credentials would be loaded twice.

I modified the has_credentials helper to support passing in api_credentials via function parameter and implemented it in 2424fcb

Reply:
This is now using the helper.